master/pkg/model/logging_config.go

```diff
@@ -82,6 +84,35 @@ func (o *ElasticSecurityConfig) Resolve() error {
 	return o.TLS.Resolve()
 }
 
+// LogRetentionPolicy configures the default log retention policy for trials and tasks.
+type LogRetentionPolicy struct {
+	// Days is the default number of days to retain logs for.
+	Days *int16 `json:"days"`
+	// Schedule is a time duration or cron expression interval to cleanup logs.
+	Schedule *string `json:"schedule"`
+}
+
+var (
+	errLogRetentionDaysParse     = errors.New("log retention days must be between -1 and 32767")
+	errLogRetentionScheduleParse = errors.New("log retention schedule must be a valid duration or cron expression")
+)
+
+// Validate implements the check.Validatable interface.
+func (p LogRetentionPolicy) Validate() []error {
+	var errs []error
+	if p.Days != nil && *p.Days < -1 {
+		errs = append(errs, errLogRetentionDaysParse)
+	}
+	if p.Schedule != nil {
+		if _, err := time.ParseDuration(*p.Schedule); err != nil {
+			if _, err := cron.ParseStandard(*p.Schedule); err != nil {
+				errs = append(errs, errLogRetentionScheduleParse)
```

Comment:
now that i'm coming back to this code for the first time in a while, i think i prefer this library to the go-cron one. seems easier to reason about the below block of code and write tests for it than the lib.

```
for {
    now := time.Now()
    next := s.Next(now)
    wait := next.Sub(now)
    time.Sleep(wait)

    do()
}
```
wdyt?

Reply:
This might be a good way to go about it. `go-co-op/gocron/v2` uses `robfig/cron/v3` under the hood. This function is kind of reimplementing robfig's `c.Run()` and would be helpful to use a fake clock for testing. It might be a good deal of re-working the code so going to leave that as an exercise for another dev due to other obligations.